sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedFlux.java

```diff
@@ -64,21 +65,71 @@ public PagedFlux(Supplier<Mono<PagedResponse<T>>> firstPageRetriever) {
      */
     public PagedFlux(Supplier<Mono<PagedResponse<T>>> firstPageRetriever,
                      Function<String, Mono<PagedResponse<T>>> nextPageRetriever) {
-        super(firstPageRetriever, nextPageRetriever);
+        this(() -> (continuationToken, pageSize) -> continuationToken == null
+            ? firstPageRetriever.get().flux()
+            : nextPageRetriever.apply(continuationToken).flux(), true);
+    }
+
+    /**
+     * PRIVATE CONSTRUCTOR. SEE BELOW NOTES.
+     *
+     * Create PagedFlux backed by Page Retriever Function Supplier.
+     *
+     * @param provider the Page Retrieval Provider
+     * @param ignored param is ignored, exists in signature only to avoid conflict with first ctr
+     */
+    // NOTES: Proposal for azure-core-v2:
+    //
+    // 1. Remove the first ctr "PagedFlux(Supplier<Mono<PagedResponse<T>>>)".
+    // 2. Add a new ctr        "PagedFlux(Supplier<PageRetriever<String, PagedResponse<T>>>)".
+    // 3. Remove the factory method "PagedFlux::create" and this PRIVATE ctr in favour of #2.
+    //
+    private PagedFlux(Supplier<PageRetriever<String, PagedResponse<T>>> provider, boolean ignored) {
```

Comment:
It's too bad that we have to use a factory to create this. I think it makes it less discoverable since you can do `new Paged(...`, and if you want to use the cooler PageRetriever, figure out that instead of `new`, use `.create`

Reply:
I wish Java has a way to work around  type erasure which is causing this pain. As pointed in the code comments - proposal is to update the current ctr taking `Mono Supplier` with `PageRetriever Supplier`, but we've to wait for major version bump up for doing it.

Reply:
Agree. Maybe this can be tracked in a github issue with appropriate labels so we don't forget to make this change when working on a major version release.